### PR TITLE
Upgrade to new build 0.4.0

### PR DESCRIPTION
**Pull request by mechanomi**
upgrade the docs to the 0.4.0 build. this includes a Mergify config

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        method: rebase
+        rebase_fallback: null
+        strict: true
+    conditions:
+      - label=ready-to-merge
+      - '#approved-reviews-by>=1'
+      - status-success=continuous-integration/travis-ci
+      - status-success=docs/readthedocs.org:crate-pdo
+    name: default
+  - name: Delete branch after merge
```
Comment (seut) [inline .mergify.yml:14]: shall we use the github option instead? we experienced a bit faster responses and reliability on the crate repository.

**Review by seut — APPROVED** [review]
Lgtm otherwise

File: .mergify.yml (hunk 1 above)
Comment (mechanomi) [reply .mergify.yml:14]: oh could do. I was copying this from elsewhere. I will remove

Comment (mechanomi) [reply .mergify.yml:14]: thanks!

**Comment by mechanomi** [comment]
@seut I pushed a fix up that does the following:

- adds a new GitHub Actions workflow to test the docs
- remove the docs tests from the Travis CI config (but leaves the code tests in place)
- updates the Mergify configuration

can you please re-review? thx!! <3

additionally, the branch protections for `master` need to be updated to:

- [ ] require the new GitHub Actions docs.yml workflow ("Build docs on macos-latest" and "Build docs on ubuntu-latest")

can you do that for me? I don't have the permissions

**Review by autophagy — APPROVED** [review]

**Comment by autophagy** [comment]
I cant see those options on the branch protection's list, but I'll check after this is merged.

**Pull request merged**